fala/apps/adviser/views.py

```diff
@@ -37,3 +38,36 @@ def get(self, request, *args, **kwargs):
 
 class AccessibilityView(TemplateView):
     template_name = "adviser/accessibility-statement.html"
+
+
+class SearchView(ListView):
+    def get(self, request, *args, **kwargs):
+        self._form = AdviserSearchForm(data=request.GET or None)
+        self._data = self._form.search()
+        return super().get(self, request, *args, **kwargs)
+
+    def get_template_names(self):
+        if self._data:
+            template = "adviser/results.html"
+        else:
+            template = "adviser/search.html"
+        return [template]
+
+    def get_queryset(self):
+        return self._data.get("results", None)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self._data:
+            pages = LaaLaaPaginator(self._data["count"], 10, 3, self._form.current_page)
+            params = {
+                "postcode": self._form.cleaned_data["postcode"],
+                "name": self._form.cleaned_data["name"],
+            }
+            categories = list(map(lambda c: "categories=" + c, self._form.cleaned_data["categories"]))
```

Comment:
I would be lying if I said I could read this line 😅

Reply:
I've added a comment to help

Reply:
It doesn't help a lot, I'm afraid. I still struggle to predict what this output even looks like 🥴 
Is it like `["categories=a", "categories=b"]`?

If I can understand what it's doing, I may be able to suggest a more readable way of doing it